### PR TITLE
Switch to jetstream

### PR DESCRIPTION
**Pull request by FestiveKyle**
Things to do before merging:
- [ ] Compare speed,error rates, and domain scan results for how the old scanners work again new type with multiple concurrency numbers (on Alpha). (Concurrency does not have to be power of 2, used just as an example). Grab JSON results from database for domain, dnsScan, and webScans after each test for comparing.
  - [x] Alpha
    - [x] Old
    - [x] 1x
    - [x] 2x
    - [x] 4x
    - [x] 8x
- [ ] Check memory/cpu usage of scanners in case we have to update any requests/limits

### CODE REVIEW

File: scanners/dns-scanner/service.py [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
 
+import nats
 from arango import ArangoClient
 from dotenv import load_dotenv
-from concurrent.futures import TimeoutError
-from dns_scanner.dns_scanner import scan_domain
-import nats
+from nats.js import JetStreamContext
+from nats.js.api import RetentionPolicy, ConsumerConfig, AckPolicy
+from nats.errors import TimeoutError
```
Comment (lcampbell2) [inline scanners/dns-scanner/service.py:19]: unused import - see line 274. Maybe import with alias (NatsTimeoutError)?

File: scanners/web-processor/service.py [hunk 2]
```diff
@@ -1,28 +1,53 @@
-import functools
+import concurrent
 import json
 import logging
 import asyncio
 import os
 import signal
+import time
+from dataclasses import dataclass
+from concurrent.futures import ThreadPoolExecutor
```
Comment (lcampbell2) [inline scanners/web-processor/service.py:9]: unused import - see line 350

File: scanners/web-processor/service.py [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
+
 import sys
 import traceback
 import re
 
-from arango import ArangoClient
+from arango import ArangoClient, DocumentUpdateError
 from dotenv import load_dotenv
 import nats
+from nats.errors import TimeoutError
```
Comment (lcampbell2) [inline scanners/web-processor/service.py:18]: unused import - see line 371

File: scanners/web-scanner/service.py [hunk 4]
```diff
@@ -1,152 +1,265 @@
...
 from dotenv import load_dotenv
-from concurrent.futures import TimeoutError, ProcessPoolExecutor
+from concurrent.futures import TimeoutError
+
+from nats.js import JetStreamContext
+from nats.js.api import RetentionPolicy, AckPolicy, ConsumerConfig
+
 from scan.web_scanner import scan_web
 import nats
+from nats.errors import TimeoutError
```
Comment (lcampbell2) [inline scanners/web-scanner/service.py:21]: unused import - see line 233. import as alias?

**Review by lcampbell2 — CHANGES_REQUESTED** [review]

File: azure-defender-easm/add-domain-to-easm/service.py [hunk 5]
```diff
@@ -8,47 +10,110 @@
...
+        "name": "SCANS",
+        "subjects": [
+            "scans.requests",
+            "scans.discovery",
+            "scans.add_domain_to_easm",
+            "scans.dns_scanner_results",
+            "scans.dns_processor_results",
+            "scans.web_scanner_results",
+            "scans.web_processor_results",
+        ],
```
[the remark this answers is not in the document]
Comment (FestiveKyle) [reply azure-defender-easm/add-domain-to-easm/service.py:71]: Just need to add them all for setup. Adding the stream fails if the configs don't match to the current state.

[the remark this answers is not in the document]
Comment (lcampbell2) [reply azure-defender-easm/add-domain-to-easm/service.py:71]: :+1:

**Review by lcampbell2 — APPROVED** [review]

**Pull request merged**